src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -157,6 +157,10 @@ public class NodeSettings : IDisposable
 
                 this.Logger.LogDebug("Network set to '{0}'.", this.Network.Name);
             }
+            else
```

Comment:
Good point @codingupastorm regarindg smart contract node startup but whilst we are here lets just do this better.

Lets go `if this.Network == null` then only create the network (i.e. we dont create and register at the same time) and after we always call `NetworkRegistration.Register(network);`

Does this make sense?

Reply:
I don't even like this ugly if condition, maybe network should just be mandatory in the constructor?

Reply:
Sounds good guys I'll do it as @fassadlr suggested